internal/collect/engine.go

```
@@ -0,0 +1,537 @@
+	if raw == "" {
+		return nil, cur, fmt.Errorf("collect %s: no list binding (items/comments/users/members) declared", name)
+	}
+	bp, err := contracts.ParsePath(raw)
+	if err != nil {
+		return nil, cur, fmt.Errorf("collect %s: bad binding %q: %w", name, raw, err)
+	}
+	var out []map[string]any
+	ccur := cur
+	pages := 0
+	for {
+		if limit > 0 && len(out) >= limit {
```

golangci-lint (2.12.2)

[error] QF1006: could lift into loop condition

(staticcheck)